src/core_codemods/with_threading_lock.py

```diff
@@ -0,0 +1,51 @@
+import libcst as cst
+from codemodder.codemods.base_codemod import ReviewGuidance
+from codemodder.codemods.api import SemgrepCodemod
+
+
+class WithThreadingLock(SemgrepCodemod):
+    NAME = "bad-lock-with-statement"
+    SUMMARY = "Replace deprecated usage of threading.Lock context manager"
+    REVIEW_GUIDANCE = ReviewGuidance.MERGE_AFTER_CURSORY_REVIEW
+    DESCRIPTION = "Separates threading lock instantiation and call with `with` statement into two steps."
+
+    @classmethod
+    def rule(cls):
+        return """
+        rules:
+          - patterns:
+            - pattern: |
+                with $BODY:
+                    ...
+            - metavariable-pattern:
+                metavariable: $BODY
+                patterns:
+                - pattern: threading.Lock()
+            - pattern-inside: |
+                import threading
+                ...
+            - focus-metavariable: $BODY
+        """
+
+    def leave_With(self, original_node: cst.With, updated_node: cst.With):
+        if original_node.items and self.node_is_selected(original_node.items[0]):
+            # TODO: how to avoid name conflicts here?
```

Comment:
You can use `ScopeProvider` to gather all the name assignments within the current, ancestors and descendants scopes. You want to make sure you don't shadow anything before (ancestors and current scopes) and are not shadowed by anything after (current and descendant scopes). 

It's easier than it seems but if you don't want to deal with it, just open a ticket and I'll look into it later.

Reply:
That's great, I think I'll make a ticket for it. Even if we can avoid collisions we still need to decide on how to generate a good name.